src/devices/Adxl345/Adxl345.cs

```diff
@@ -0,0 +1,173 @@
+        private const byte ADDRESS_Y0 = 0x34;               // Address of the Y Axis data register              
+        private const byte ADDRESS_Z0 = 0x36;               // Address of the Z Axis data register               
+
+        private const byte ACCEL_SPI_RW_BIT = 0x80;         // Bit used in SPI transactions to indicate read/write  
+        private const byte ACCEL_SPI_MB_BIT = 0x40;         // Bit used to indicate multi-byte SPI transactions    
+        #endregion
+
+        /// <summary>
+        /// Read Acceleration from ADXL345
+        /// </summary>
+        public Vector3 Acceleration => ReadAcceleration();
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="busId">SPI Bus ID</param>
+        /// <param name="chipSelect">CS Pin</param>
+        /// <param name="gravityRange">Gravity Measurement Range</param>
+        public Adxl345(int busId, int chipSelect, GravityRange gravityRange)
+        {
+            if (gravityRange == GravityRange.Range1)
+            {
+                _range = 4;
+            }
+            else if (gravityRange == GravityRange.Range2)
+            {
+                _range = 8;
+            }
+            else if (gravityRange == GravityRange.Range3)
+            {
+                _range = 16;
+            }
+            else if (gravityRange == GravityRange.Range4)
+            {
+                _range = 32;
+            }
+            _gravityRangeByte = (byte)gravityRange;
+
+            _busId = busId;
+            _chipSelect = chipSelect;
+            var settings = new SpiConnectionSettings(_busId, _chipSelect)
+            {
+                ClockFrequency = 5000000,
+                Mode = SpiMode.Mode3
+            };
+            _sensor = RuntimeInformation.IsOSPlatform(OSPlatform.Linux) ? (SpiDevice)new UnixSpiDevice(settings) : new Windows10SpiDevice(settings);
+
+            Initialize();
+        }
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="sensor">The communications channel to a device on a SPI bus</param>
+        /// <param name="gravityRange">Gravity Measurement Range</param>
+        public Adxl345(SpiDevice sensor, GravityRange gravityRange)
+        {
+            if (gravityRange == GravityRange.Range1)
+            {
+                _range = 4;
+            }
+            else if (gravityRange == GravityRange.Range2)
+            {
+                _range = 8;
+            }
+            else if (gravityRange == GravityRange.Range3)
+            {
+                _range = 16;
+            }
+            else if (gravityRange == GravityRange.Range4)
+            {
+                _range = 32;
+            }
+            _gravityRangeByte = (byte)gravityRange;
+
+            _sensor = sensor;
+
+            Initialize();
+        }
+
+        /// <summary>
+        /// Initialize ADXL345
+        /// </summary>
+        internal void Initialize()
```

Comment:
Why internal and not private? where else do you need to call this from?

Reply:
I was wrong. I was wondering if I could see this outside the assembly.😅